tests/test_cadsource.py

```diff
@@ -23,23 +25,24 @@ def test_CADSource_to_dict():
 
 
 def test_CADSource_to_dict_absolute():
-    x = CADSource(name="Rapido Hotend", path=Path("/tmp/abc/xyz.step"))
+    x = CADSource(name="Rapido Hotend", path=Path("C:/tmp/abc/xyz.step"))
+    # In Linux, /tmp/abc/xyz.step would be an absolute path, 
+    # but on Windows, an absolute path would look like C:\tmp\abc\xyz.step
 
     # note the provision of the optional root path so that to_dict()
     # can return the appropriate relative path
     d = x.to_dict(root=Path("/tmp"))
     assert isinstance(d, dict)
     assert d["type"] == "CADSource"
-    assert d["name"] == name
+    assert d["name"] == x.name
     assert d["path"] == "abc/xyz.step"  # note: this is not a path, but a string
 
     p = Path(d["path"])
     assert not p.is_absolute()
 
 
 def test_CADSource_to_dict_absolute_error():
-    x = CADSource(name="Rapido Hotend", path=Path("/tmp/abc/xyz.step"))
-
+    x = CADSource(name="Rapido Hotend", path=Path("C:/tmp/abc/xyz.step")) 
```

Comment:
Windows-specific change.

Reply:
I've been thinking about this, and I think we need two tests -- one for Unix-like systems and another for Windows-like systems. The Windows tests can add a drive letter, etc.